keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ApplicationController.kt

```diff
@@ -49,6 +49,7 @@ class ApplicationController(
     path = ["/{application}"],
     produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
   )
+  @PreAuthorize("@authorizationSupport.userCan('READ', 'APPLICATION', #application)")
```

Comment:
Example of what calling the new `userCan` function looks like.

Reply:
why is there a new way to do this here when everywhere else in Spinnaker this would be

` @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ')")`

Reply:
I just looked in authorization support, so I get what is going on in there now and it makes a bit more sense, I'm still not sure that for some of these checks they shouldn't just follow the standard model, looking a bit more

Reply:
This was originally done this way because we didn't have all the info in the API call that we needed to look up the permissions. I had to do some additional querying. I bet we could follow the standard model for some of them/maybe we could have done it all along

Reply:
I think it is worth a discussion on ApplicationController what is required to view/pause/resume management (I'm not totally sure it is the same as having access to manage the resource configuration of the application / needs to be gated on access to the service account)

Reply:
for example if I can see your application but can't modify it, I should maybe be able to tell it is a managed application?

Reply:
@cfieber I had the same reaction when initially working on this, and that was one of the reasons I refactored it to be as close as possible to what the other services already use. But as you pointed out, there's some additional logic to find out the service account, and after my changes also the application name, so we can actually call the permission evaluator. If you can think of an easier/better way to do this, I'm all 👂s.

Reply:
I think if I understand correctly that everything funnelling through userCan is a test on whether the user has permission to the service account for the entity and that the service account has the permission being tested.

I think that only accounts for the persona of an application owner that can manage the delivery configuration for the application (they have permission to the service account).

I suspect (in application controller at least) that access to details on whether the application is under declarative management would be relevant to anyone who has permission to view the application.

I'd just propose going through each of the places where auth has been applied and considering the security annotation under the lens of what specifically that endpoint is serving and make sure we haven't locked other users out